### PR TITLE
Query and methods bug fixes and documentation refactoring

### PR DESCRIPTION
**Pull request by iK4tsu**
Fixed:
 * Query when built with 1 component and a filter was storing the wrong components, leading to a wrong output.
 * Multiple functions were lacking attributes.
 * Remove alias this from Entity, its intended behavior being overshadowed by opEquals making it unused.
 * Fixed removeIfHas method in EntityManager not safely evaluating an entity.
 * Refactor entities and components methods in Storage to package visibility.
 * Added signal module to package.d

### CODE REVIEW

**Review by Ev1lbl0w — APPROVED** [review]
LGTM

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[*]
+insert_final_newline = true
+indent_size = 4
+indent_style = tab
+tab_width = 4
```
Comment (ljmf00) [inline .editorconfig:5]: You could add `trim_trailing_whitespace = true` to trim useless trailing whitespaces.

Comment (iK4tsu) [reply .editorconfig:5]: done

**Comment by iK4tsu** [comment]
Also updated README.md

**Review by ljmf00 — APPROVED** [review]

**Pull request merged**